src/Item_Line.php

```diff
@@ -0,0 +1,576 @@
+                }
+                echo "<td>" . $line->getLink() . "</td>";
+                echo "</tr>";
+            }
+            echo $header;
+            echo "</table>";
+
+            if ($canedit && count($lines)) {
+                $massiveactionparams['ontop'] = false;
+                Html::showMassiveActions($massiveactionparams);
+            }
+            if ($canedit) {
+                Html::closeForm();
+            }
+        }
+    }
+
+    public function showForm($ID, array $options = [])
+    {
+        global $DB, $CFG_GLPI;
+
+        echo "<div class='center'>";
+
+        $this->initForm($ID, $options);
+        $this->showFormHeader();
+
+        $line = new Line();
+        $line->getFromDB($this->fields['lines_id']);
+
+        $rand = mt_rand();
+
+        echo "<tr class='tab_bg_1'>";
+        echo "<td><label for='dropdown_itemtype$rand'>" . __('Item type') . "</label></td>";
+        echo "<td>";
+        $types = $CFG_GLPI['line_types'];
+        $translated_types = [];
+        foreach ($types as $type) {
+            $translated_types[$type] = $type::getTypeName(1);
+        }
+        Dropdown::showFromArray(
+            'itemtype',
+            $translated_types,
+            [
+                'display_emptychoice'   => true,
+                'value'                 => $this->fields["itemtype"],
+                'rand'                  => $rand
+            ]
+        );
+
+        //get all used items
+        $used = [];
+        $iterator = $DB->request([
+            'FROM'   => $this->getTable()
+        ]);
+        foreach ($iterator as $row) {
+            $used [$row['itemtype']][] = $row['items_id'];
+        }
+
+        Ajax::updateItemOnSelectEvent(
+            "dropdown_itemtype$rand",
+            "items_id",
+            $CFG_GLPI["root_doc"] . "/ajax/dropdownAllItems.php",
+            [
+                'idtable'   => '__VALUE__',
+                'name'      => 'items_id',
+                'value'     => $this->fields['items_id'],
+                'rand'      => $rand,
+                'used'      => $used
+            ]
+        );
+
+        echo "</td>";
+        echo "<td><label for='dropdown_items_id$rand'>" . _n('Item', 'Items', 1) . "</label></td>";
+        echo "<td id='items_id'>";
+        if (isset($this->fields['itemtype']) && !empty($this->fields['itemtype'])) {
+            $itemtype = $this->fields['itemtype'];
+            $itemtype = new $itemtype();
+            $itemtype::dropdown([
+                'name'   => "items_id",
+                'value'  => $this->fields['items_id'],
+                'rand'   => $rand
+            ]);
+        } else {
+            Dropdown::showFromArray(
+                'items_id',
+                [],
+                [
+                    'display_emptychoice'   => true,
+                    'rand'                  => $rand
+                ]
+            );
+        }
+
+        echo "</td>";
+        echo "</tr>";
```

Comment:
Maybe this whole part can be replaced by usage of `Dropdown::showSelectItemFromItemtypes()`